docs/CI-PIPELINES.md

```diff
@@ -0,0 +1,420 @@
+        // log on failure, but propagate the error outward.
+        sh './build.sh'
+    }
+    catch(e) {
+        archiveArtifacts allowEmptyArchive: true, artifacts: "msbuild.log"
+        throw e
+    }
+}
+```
+## Multi-machine workflows and stashing
+One of the strengths of the Jenkins pipelines is the ability to orchestrate multiple machines in the same script. To do this, we simply write a pipeline that has multiple node blocks.  We can pass files between blocks any way we'd like (e.g. Azure Blob Storage), but one easy way is to use the `stash` and `unstash` steps to create a small temporary archive. Note that caution should be used to limit archive size.
+```
+simpleNode('Windows_NT','latest') {
+    // Build some stuff
+    bat 'build.cmd'
+    // Stash this for a subsequent Linux job
+    stash name: 'nt-build-artifacts', includes: 'bin/foo.dll'
+}
+
+simpleNode('Ubuntu14.04','latest') {
+    dir ('linux-bin') {
+        // Unstash to a different directory
+        unstash 'nt-build-artifacts'
+    }
+    // Build some stuff
+    sh './build.sh linux-bin/foo.dll'
+}
+```
+
+## .NET CI SDK Utility Functionality
+The .NET CI SDK, which is imported at the start of your pipelines, contains a number of utility steps, sometimes called 'global variables' in pipelines, which make life easier.  Each step is implemented in its own script file under the [vars](../vars) directory in this repository.  Most utility steps should be fairly self explanatory.  Please see the [defining global variables](https://jenkins.io/doc/book/pipeline/shared-libraries/#defining-global-variables) for info on how they work.  Some interesting ones are highlighted here
+* getUser() - Attempts to get the user who started the run.  Can be useful for some reporting/submission steps
+* getCommit() - Attempts to get the commit for this run.  For a PR, this will be the PR source commit.  For a push/manually triggered run it's the commit of the repo in the directory which getCommit() is called.  This is becuase a job can check out multiple repositories
+* getLogFolder() - Gets the name of the log folder automatically created in the workspace when `simpleNode` or `simpleDockerNode` is used.  This folder is automatically archived before existing the node.
+* getHelixSource() - For Helix submissions, attempts to construct a helix source for the run that can be passed to the Helix submission script.
+* waitForHelixRuns(helixRunsBlob) - Given the json output of SubmittedHelixRuns.txt, wait for all of the runs to finish and report their status.  It does not need to be called inside a node block, and this should be avoided to improve efficiency.
+    ```
+    def logFolder = getLogFolder()
+    def submittedHelixJson = readJSON file: "${logFolder}/SubmittedHelixRuns.txt"
+    waitForHelixRuns(submittedHelixJson)
+    ```
+## .NET CI Pipeline Examples
+
+* [CoreFX Linux pipeline](https://github.com/dotnet/corefx/blob/master/buildpipeline/portable-linux.groovy).
```

Comment:
This link is not working for me. The file is not there. Are you planning on adding it? or did it change?